skfda/_utils/_utils.py

```diff
@@ -564,6 +570,35 @@ def _classifier_get_classes(y):
     classes = le.classes_
 
     if classes.size < 2:
-        raise ValueError(f'The number of classes has to be greater than'
-                         f' one; got {classes.size} class')
+        raise ValueError(
+            f'The number of classes has to be greater than'
+            f'one; got {classes.size} class',
+        )
     return classes, y_ind
+
+
+def _classifier_get_distributions(
+    classes: ndarray,
+    X: FDataGrid,
+    y_ind: ndarray,
+    depth_methods: Sequence[Depth],
+) -> Sequence[Depth]:
+    return [
+        clone(depth_method).fit(X[y_ind == cur_class])
+        for cur_class in range(classes.size)
+        for depth_method in depth_methods
+    ]
+
+
+def _classifier_fit_distributions(
+    X: FDataGrid,
+    y: ndarray,
+    depth_methods: Sequence[Depth],
```

Comment:
Depth should be parameterized with a TypeVar in both parameter and return types.